cmd/kube-proxy/app/server.go

```diff
@@ -446,7 +446,11 @@ with the apiserver API to configure the proxy.`,
 			if err := opts.Validate(args); err != nil {
 				klog.Fatalf("failed validate: %v", err)
 			}
-			klog.Fatal(opts.Run())
+
+			if err := opts.Run(); err != nil {
+				klog.Error(err)
+				os.Exit(1)
```

Comment:
You can just do `klog.Exit(err)`, which will exit without stacktracing. I agree that the stack trace is unwanted in this context.

Reply:
Changed the code. Thanks for the hint !